ast/parser.go

```diff
@@ -30,6 +30,8 @@ var RegoV1CompatibleRef = Ref{VarTerm("rego"), StringTerm("v1")}
 // RegoVersion defines the Rego syntax requirements for a module.
 type RegoVersion int
 
+const DefaultRegoVersion = RegoVersion(0)
+
 const (
 	// RegoV0 is the default, original Rego syntax.
 	RegoV0 RegoVersion = iota
```

Comment:
If we change the order of this enum, we effectively set the default rego-version to whatever entry is the first (`= iota`).
If I change this to 

```go
const (
	// RegoV1 is the Rego syntax enforced by OPA 1.0; e.g.:
	// future.keywords part of default keyword set, and don't require imports;
	// 'if' and 'contains' required in rule heads;
	// (some) strict checks on by default.
	RegoV1 RegoVersion = iota
	// RegoV0CompatV1 requires modules to comply with both the RegoV0 and RegoV1 syntax (as when 'rego.v1' is imported in a module).
	// Shortly, RegoV1 compatibility is required, but 'rego.v1' or 'future.keywords' must also be imported.
	RegoV0CompatV1
	// RegoV0 is the default, original Rego syntax.
	RegoV0
)
```

Then some tests in `ast/parser_test.go` starts to fail, e.g.:

* https://github.com/ashutosh-narkar/opa/blob/25d21f50d358cbb31f47ee0f00b2d006a6068e13/ast/parser_test.go#L663
* https://github.com/ashutosh-narkar/opa/blob/25d21f50d358cbb31f47ee0f00b2d006a6068e13/ast/parser_test.go#L796
* https://github.com/ashutosh-narkar/opa/blob/25d21f50d358cbb31f47ee0f00b2d006a6068e13/ast/parser_test.go#L1233
* https://github.com/ashutosh-narkar/opa/blob/25d21f50d358cbb31f47ee0f00b2d006a6068e13/ast/parser_test.go#L1344
* https://github.com/ashutosh-narkar/opa/blob/25d21f50d358cbb31f47ee0f00b2d006a6068e13/ast/parser_test.go#L1947
* https://github.com/ashutosh-narkar/opa/blob/25d21f50d358cbb31f47ee0f00b2d006a6068e13/ast/parser_test.go#L2888
* https://github.com/ashutosh-narkar/opa/blob/25d21f50d358cbb31f47ee0f00b2d006a6068e13/ast/parser_test.go#L3242
* https://github.com/ashutosh-narkar/opa/blob/25d21f50d358cbb31f47ee0f00b2d006a6068e13/ast/parser_test.go#L3294
* https://github.com/ashutosh-narkar/opa/blob/25d21f50d358cbb31f47ee0f00b2d006a6068e13/ast/parser_test.go#L3453
* https://github.com/ashutosh-narkar/opa/blob/25d21f50d358cbb31f47ee0f00b2d006a6068e13/ast/parser_test.go#L3492

Reply:
> If we change the order of this enum, we effectively set the default rego-version to whatever entry is the first (= iota).

Interesting. Thanks for the pointer. I'll work on fixing those tests.